packages/salesforcedx-vscode-soql/src/commands/soqlBuilderToggle.ts

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import * as vscode from 'vscode';
+import {
+  BUILDER_VIEW_TYPE,
+  EDITOR_VIEW_TYPE,
+  OPEN_WITH_COMMAND
+} from '../constants';
+import { telemetryService } from '../telemetry';
+
+export async function soqlBuilderToggle(doc: vscode.Uri): Promise<void> {
+  telemetryService.sendCommandEvent('soql_builder_toggle', process.hrtime());
+
+  const viewType = vscode.window.activeTextEditor
+    ? BUILDER_VIEW_TYPE
+    : EDITOR_VIEW_TYPE;
```

Comment:
It's not clear to me how this works for us, is `vscode.window.activeTextEditor` saying, 
"if the active editor is of type `text editor` then return true." ?

Is the there a `vscode.window.activeCustomEditor`?

Reply:
Basically, vscode.window.activeTextEditor is either undefined or an object.  It's an object if the soql source is displayed in the text editor.  It's undefined if the custom editor is being displayed.  So I'm basically setting viewType to either our custom editor viewType `soqlCustom.soql` or `default` ( which is always the text editor ).

There is no such thing as a `vscode.window.activeCustomEditor`...sure wish they had one.

Reply:
> viewType to either our custom editor viewType soqlCustom.soql or default ( which is always the text editor ).

@dehru @praksb The text editor is the default view for a `.soql` file because we have `"priority": "option"` in the package.json. However, the user does have the option to change that in the settings. If they due then I think `EDITOR_VIEW_TYPE` might also be `soqlCustom.soql`.

Did you text the case here when the "default" is no longer the text editor?

Reply:
![image](https://user-images.githubusercontent.com/25188632/103016941-1b950680-4500-11eb-822c-c33b86af8ed7.png)

Reply:
I'm not sure if the `default` value used in the ternary points to whatever the default this setting refers too or `default` is just the VS Code default editor as if there is no alternative.

Reply:
I was following up on the issue with tabs I made in the VS Code repo and the MS liked it to a [duplicate](https://github.com/microsoft/vscode/issues/111474) that might be useful here. Did you see the command `View: Toggle editor type` in the command palette? Maybe that will handle the view type for us regardless of any settings and defaults.

Reply:
@jonnyhork 
I just now tested the case by switching the SOQL Builder view to be default and it works fine. 
I also tried  using View: Toggle editor type in the command palette and it looks like it should work but currently when I use it I am running into the VS Code bug that you mentioned above.

Reply:
Thanks for testing it, I'm glad it works as is!